src/units/KM_Units.pas

```diff
@@ -307,6 +313,79 @@ function TKMUnitCitizen.FindHome: Boolean;
 end;
 
 
+procedure TKMSettledUnit.CleanHousePointer(aFreeAndNilTask: Boolean = False);
+begin
+  if aFreeAndNilTask then
+    FreeAndNil(fUnitTask);
+  fHome.GetHasOwner := False;
+  gHands.CleanUpHousePointer(fHome);
+end;
+
+// Manage house is closed for worker process
+procedure TKMSettledUnit.ProceedHouseClosedForWorker;

```

Comment:
Hahah that's understandable. Our unit task/action code is quite fragile, you need to know about how all the tasks/actions work in lots of places. It's one of the oldest parts of the code base and not perfectly designed :)